### PR TITLE
[MBL-16785][Student] Offline Module list

### PR DESCRIPTION
**Pull request by tamaskozmer**
Test plan: Test the module list in offline and online mode. Try to test pagination as well. Lot of refactoring happened on the Module screen so it should be well tested in online mode as well.

refs: MBL-16785
affects: Student
release note: none

## Checklist

- [x] Run E2E test suite
- [x] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/student/src/main/java/com/instructure/student/features/modules/list/adapter/ModuleListRecyclerAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+            }
+        }.start()
+    }
+
+    private fun handleModuleItemResponse(result: DataResult<List<ModuleItem>>, moduleObject: ModuleObject, isNotifyGroupChange: Boolean) {
+        if (result is DataResult.Success) {
+            val resultIsFromApiOrDb = result.apiType === ApiType.API || result.apiType == ApiType.DB
+            val moduleItems = result.data
+
+            var position = if (moduleItems.isNotEmpty() && moduleItems[0] != null) moduleItems[0].position - 1 else 0
```
[the remark this answers is not in the document]
Comment (tamaskozmer) [reply apps/student/src/main/java/com/instructure/student/features/modules/list/adapter/ModuleListRecyclerAdapter.kt:234]: Even though that is not a real change, Github just indicates that this whole file is new, but it's still a good point.

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Pull request merged**